treeherder/webapp/api/jobs.py

```diff
@@ -147,6 +147,20 @@ def retrigger(self, request, project, jm, pk=None):
         else:
             return Response("No job with id: {0}".format(pk), 404)
 
+    @action(permission_classes=[IsAuthenticated])
```

Comment:
There is already a class defined in permissions.py: https://github.com/mozilla/treeherder/blob/master/treeherder/webapp/api/permissions.py#L8
I will use that.